File: Test/Amazon2.py
def getMinimumCost(parcels, k):
    n = len(parcels)
    min_sum = 0
    if(n==k): return min_sum
    i = 1
    num_rem = k-n

    checker = set()

    for j in parcels:
        checker.add(j)
    print(checker)

    while(num_rem > 0):
        if(not (i in checker)):
            min_sum += i
            num_rem -= 1
        i += 1
    return min_sum

    
        

## test Case 1 || Should output 17
parcels = [2,3,6,10,11]
k = 9

## test Case 2 || Should output 9
parcels1 = [6,5,4,1,3]
k1 = 7

## test Case 3 || SHould output 0
parcels2 = [4,5,7,1]
k2 = 4

print(getMinimumCost(parcels, k))

File: Test/Amazon2.java
package Interviews;

import java.util.*;
public class Amazon2 {
    public static int getMinimum(int[] parcels, int k){
        List<Integer> parcels_lst = new ArrayList<>();
        for(int parcel : parcels){
            parcels_lst.add(parcel);
        }

        int i = 1;
        int sum = 0;
        while(parcels_lst.size() != k){
            if(!parcels_lst.contains(i)){
                parcels_lst.add(i);
                sum += i;
            }
            i += 1;
        }
        return sum;
    }

    public static void main(String[] args) {
        //Test Case 1
        int[] parcels = {2,3,6,10,11};
        int k = 9;

        // //Test Case 2
        // int[] parcels1 = {6,5,4,1,3};
        // int k1 = 7;

        // //Test Case 3
        // int[] parcels2 = {4,5,7,1};
        // int k2 = 4;
        
        System.out.println(getMinimum(parcels, k));
        
    }
    
}
